client/client.go

```diff
@@ -120,14 +125,83 @@ func (h *HTTPClient) RunQuery(ctx context.Context, query string) ([]byte, error)
 
 	defer resp.Body.Close()
 
-	body, err := ioutil.ReadAll(resp.Body)
+	body, err := io.ReadAll(resp.Body)
 	if err != nil {
 		return nil, fmt.Errorf("read body: %w", err)
 	}
 
 	return body, nil
 }
 
+// RefreshToken performs a refresh token request.
+// The method set the *HTTPClient.accessToken field to the new access token.
+func (h *HTTPClient) RefreshToken(ctx context.Context) error {
+	refreshTokenRequest := refreshTokenRequest{
+		RefreshToken: h.refreshToken,
+	}
+
+	buffer := bytes.NewBuffer(nil)
+	if err := json.NewEncoder(buffer).Encode(&refreshTokenRequest); err != nil {
+		return fmt.Errorf("encode refresh token request: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, refreshTokenURL, buffer)
+	if err != nil {
+		return fmt.Errorf("create refresh token request: %w", err)
+	}
+
+	req.Header.Set("Content-Type", "application/json;charset=UTF-8")
+
+	resp, err := h.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("do refresh token request: %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
```

Comment:
Did you check ,  is firebolt return 200?

Reply:
Yes, Firebolt returns `200` on success